mono/mini/mini.c

```diff
@@ -4125,6 +4125,11 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 
 	mono_domain_lock (target_domain);
 
+	if (mono_stats_method_desc && mono_method_desc_full_match (mono_stats_method_desc, method)) {
+		g_printf ("Printing JIT stats at method: %s\n", mono_method_get_full_name (method));
+		mono_runtime_print_stats ();
+	}
+
```

Comment:
Wouldn't it be better to do this lower in this function, near ```MONO_PROFILER_RAISE (jit_done, (method, jinfo));```

Reply:
I think it was located here to be within the domain lock, which is theoretically the lock you're supposed to take around the structs.